client_old/Page.ts

```diff
@@ -1,11 +1,9 @@
-import { IPageData } from '../model/Page.interface';
-
 export default class Page {
     private index: number;
     private url: string;
 
-    constructor(data: IPageData) {
-        this.index = typeof data.index === 'number' ? data.index : parseInt(data.index, 10);
+    constructor(data) {
+        this.index = parseInt(data.index);
```

Comment:
It's a bad idea to remove the radix: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Description